### PR TITLE
Carousel: Separate draggable area from controls

### PR DESCRIPTION
**Pull request by Mitch-At-Work**
## Previous Behavior
When 'draggable' was enabled on carousel the draggable area included anything in the main Carousel wrapper

## New Behavior
Draggable now only enables draggable area within the CarouselSlider window

This aligns with the reccomended Embla pattern for draggable carousels:
https://www.embla-carousel.com/guides/previous-and-next-buttons/

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

File: change/@fluentui-react-carousel-2e2a03df-5a80-41c4-b8df-9745b659c556.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-carousel-2e2a03df-5a80-41c4-b8df-9745b659c556.json:1]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv9 Visual Regression Report](https://aka.ms/visual-changes-review?organization=uifabric&project=fabricpublic&prId=33155&commitId=aa5a7dbb8d17554a3e40f1786e8b224537ddc4af&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.Badge Mask RTL.chromium.png|21| Changed|
|Avatar Converged.badgeMask.chromium.png|30| Changed|

</details>


  <details><summary><b>Drawer</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Drawer.Full Overlay High Contrast.chromium.png|4082| Changed|
|Drawer.overlay drawer full.chromium.png|1154| Changed|
|Drawer.Full Overlay Dark Mode.chromium.png|8| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/react-components/react-carousel/stories/src/Carousel/CarouselDefault.stories.tsx [hunk 2]
```diff
@@ -77,7 +77,7 @@ const getAnnouncement: CarouselAnnouncerFunction = (index: number, totalSlides:
 };
 
 export const Default = () => (
-  <Carousel groupSize={1} circular announcement={getAnnouncement}>
+  <Carousel draggable groupSize={1} circular announcement={getAnnouncement}>
```
Comment (layershifter) [inline packages/react-components/react-carousel/stories/src/Carousel/CarouselDefault.stories.tsx:80]: Revert this?

File: packages/react-components/react-carousel/library/src/components/pointerEvents.ts [hunk 3]
```diff
@@ -32,17 +32,29 @@ export function pointerEventPlugin(options: PointerEventPluginOptions): PointerE
...
+    targetDocument.addEventListener('pointerdown', documentDownListener);
+  }
+
   function clearPointerEvent() {
     pointerEvent = undefined;
+    reinitMouseDown();
   }
 
-  function selectListener() {
+  function selectListener(_emblaApi: EmblaCarouselType, evt: EmblaEventType) {
```
Comment (layershifter) [inline packages/react-components/react-carousel/library/src/components/pointerEvents.ts:50]: Curious, why this change is needed?

File: packages/react-components/react-carousel/library/src/components/pointerEvents.ts [hunk 4]
```diff
@@ -32,17 +32,29 @@ export function pointerEventPlugin(options: PointerEventPluginOptions): PointerE
...
 
+  function reinitMouseDown() {
+    // Reinitialize the listener for next mouse/pointerDown event
+    const targetDocument = emblaApi.containerNode().ownerDocument;
+    targetDocument.removeEventListener('mousedown', documentDownListener);
+    targetDocument.removeEventListener('pointerdown', documentDownListener);
+
+    targetDocument.addEventListener('mousedown', documentDownListener);
+    targetDocument.addEventListener('pointerdown', documentDownListener);
+  }
```
Comment (layershifter) [inline packages/react-components/react-carousel/library/src/components/pointerEvents.ts:43]: This does not look correct as it just removes and immediately adds listeners

**Review by layershifter — COMMENTED** [review]
IMO the current change in the structure does not scale 😐

I played with it a bit and I propose to try the following plan:
1. Undo all changes & deprecate `CarouselSlider`
2. Create new `CarouselViewport` component that will have `root` (i.e. `embla__viewport`) & `slider` (_better name?_ i.e. `embla__container`) slots
3. Undo changes in `useEmblaCarousel()`
4. Rename `containerRef` in `useEmblaCarousel()` to `viewportRef`, add `viewportRef` to `CarouselContext` and consume it in `CarouselViewport`. This way we avoid `.querySelector()` in `useEmblaCarousel()`
5. Ensure that this still somehow works with deprecated `CarouselSlider`

File: packages/react-components/react-carousel/library/src/components/pointerEvents.ts (hunk 3 above)
Comment (Mitch-At-Work) [reply packages/react-components/react-carousel/library/src/components/pointerEvents.ts:50]: I figured it would be safer to enact on the direct _emblaApi passed in via event in case it's somehow reinit'd inbetween.

Comment (layershifter) [reply packages/react-components/react-carousel/library/src/components/pointerEvents.ts:50]: image

We should always have proper instance, it's how other plugins written, see for reference: https://github.com/davidjerleke/embla-carousel/blob/67a1ac0d8821a2eadf6232e9663fef59a21a9abd/packages/embla-carousel-autoplay/src/components/Autoplay.ts#L40-L44

File: packages/react-components/react-carousel/stories/src/Carousel/index.stories.tsx [hunk 5]
```diff
@@ -33,7 +34,7 @@ export default {
     CarouselNavButton,
     CarouselNavContainer,
     CarouselNavImageButton,
-    CarouselSlider,
+    CarouselViewport,
```
Comment (layershifter) [inline packages/react-components/react-carousel/stories/src/Carousel/index.stories.tsx:37]:
```suggestion
    CarouselSlider,
    CarouselViewport,
```

File: packages/react-components/react-carousel/stories/src/Carousel/CarouselDescription.md [hunk 6]
```diff
@@ -4,7 +4,7 @@ A `Carousel` component is a sliding window of elements controlled by previous, n
...
 
-A `CarouselCard` can be full screen, responsive, or partial sizes, it is recommended to enable the `cardFocus` prop on the `CarouselSlider` if the cards are not full width banners (this provided keyboard navigation and accessibility tool access). If cards are intended to be full-screen banners, we recommend relying on the tab index of the internal elements only, while the out-of-view cards will be set to aria-hidden by default to prevent unnecessary tabbing and quick control access.
+A `CarouselCard` can be full screen, responsive, or partial sizes, it is recommended to enable the `cardFocus` prop on the `CarouselViewport` if the cards are not full width banners (this provided keyboard navigation and accessibility tool access). If cards are intended to be full-screen banners, we recommend relying on the tab index of the internal elements only, while the out-of-view cards will be set to aria-hidden by default to prevent unnecessary tabbing and quick control access.
```
Comment (layershifter) [inline packages/react-components/react-carousel/stories/src/Carousel/CarouselDescription.md:7]:
```suggestion
A `CarouselCard` can be full screen, responsive, or partial sizes, it is recommended to enable the `cardFocus` prop on the `CarouselSlider` if the cards are not full width banners (this provided keyboard navigation and accessibility tool access). If cards are intended to be full-screen banners, we recommend relying on the tab index of the internal elements only, while the out-of-view cards will be set to aria-hidden by default to prevent unnecessary tabbing and quick control access.
```

File: packages/react-components/react-carousel/stories/src/Carousel/CarouselImageBox.stories.tsx [hunk 7]
```diff
@@ -71,13 +71,15 @@ export const ImageSlideshow = () => {
...
   return (
     <Carousel groupSize={1} align="center" announcement={getAnnouncement}>
-      <CarouselSlider className={classes.slider}>
-        {IMAGES.map((image, index) => (
-          <CarouselCard key={image.url} className={classes.card} aria-label={`${index + 1} of ${IMAGES.length}`}>
-            <ImageCard {...image} />
-          </CarouselCard>
-        ))}
-      </CarouselSlider>
+      <CarouselViewport className={classes.slider}>
```
Comment (layershifter) [inline packages/react-components/react-carousel/stories/src/Carousel/CarouselImageBox.stories.tsx:74]:
```suggestion
      <CarouselViewport className={classes.viewport}>
```

File: packages/react-components/react-carousel/stories/src/Carousel/CarouselBestPractices.md [hunk 8]
```diff
@@ -3,7 +3,7 @@
 ### Do
 
 - Limit cards to a max width of 100% of screen size.
-- Use 'gap' css on CarouselSlider to provide easy spacing between cards, use margin if circular is enabled.
+- Use 'gap' css on CarouselViewport's slider to provide easy spacing between cards, use margin if circular is enabled.
```
Comment (layershifter) [inline packages/react-components/react-carousel/stories/src/Carousel/CarouselBestPractices.md:6]:
```suggestion
- Use 'gap' css on `CarouselSlider` to provide easy spacing between cards, use margin if circular is enabled.
```

File: packages/react-components/react-carousel/library/src/components/CarouselViewport/CarouselViewport.tsx [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+  const context = useCarouselSliderContextValues_unstable(state);
+  /**
+   * @see https://github.com/microsoft/fluentui/blob/master/docs/react-v9/contributing/rfcs/react-components/convergence/custom-styling.md
+   *
+   * TODO: 💡 once package will become stable (PR which will be part of promoting PREVIEW package to STABLE),
+   *      - uncomment this line
+   *      - update types {@link file://./../../../../../../../packages/react-components/react-shared-contexts/library/src/CustomStyleHooksContext/CustomStyleHooksContext.ts#CustomStyleHooksContextValue}
+   *      - verify that custom global style override works for your component
+   */
+  // useCustomStyleHook_unstable('useCarouselViewportStyles_unstable')(state);
```
Comment (layershifter) [inline packages/react-components/react-carousel/library/src/components/CarouselViewport/CarouselViewport.tsx:25]: Let's address this

**Review by layershifter — APPROVED** [review]
